src/pages/UserPage/UserRegister.jsx

```
@@ -1,5 +1,276 @@
+    setCheckedItems((prev) => ({ ...prev, [id]: !prev[id] }));
+  };
+
+  const handleCheckAll = () => {
+    const newState = !allChecked;
+    setCheckedItems(
+      agreements.reduce((acc, { id }) => ({ ...acc, [id]: newState }), {}),
+    );
+  };
+
+  const checkNicknameDuplicate = async () => {
+    if (!form.nickname.trim()) {
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)